services/cd-service/pkg/repository/repository.go

```diff
@@ -2145,28 +2145,27 @@ func (s *State) GetAllQueuedAppVersions(ctx context.Context) (db.AllQueuedVersio
 	return result, nil
 }
 
-func (s *State) GetAllCommitEvents(ctx context.Context) (db.AllCommitEvents, error) {
-	ddSpan, _ := tracer.StartSpanFromContext(ctx, "GetAllCommitEvents")
+func (s *State) WriteAllCommitEvents(ctx context.Context, transaction *sql.Tx, dbHandler *db.DBHandler) error {
+	ddSpan, _ := tracer.StartSpanFromContext(ctx, "WriteAllCommitEvents")
 	defer ddSpan.Finish()
 	fs := s.Filesystem
 	allCommitsPath := "commits"
 	commitPrefixes, err := fs.ReadDir(allCommitsPath)
 	if err != nil {
-		return nil, fmt.Errorf("could not read commits dir: %v\n", err)
+		return fmt.Errorf("could not read commits dir: %v\n", err)
 	}
-	result := make(db.AllCommitEvents)
 	for _, currentPrefix := range commitPrefixes {
 		currentpath := fs.Join(allCommitsPath, currentPrefix.Name())
 		commitSuffixes, err := fs.ReadDir(currentpath)
 		if err != nil {
-			return nil, fmt.Errorf("could not read commit directory '%s': %v", currentpath, err)
+			return fmt.Errorf("could not read commit directory '%s': %v", currentpath, err)
```

Comment:
when wrapping errors with fmt.Errorf, please always use `w`. This creates an error that can be unwrapped
```suggestion
			return fmt.Errorf("could not read commit directory '%s': %w", currentpath, err)
```